core/chains/evm/logpoller/log_poller.go

```diff
@@ -63,6 +63,72 @@ type LogPoller interface {
 	LogsDataWordBetween(ctx context.Context, eventSig common.Hash, address common.Address, wordIndexMin, wordIndexMax int, wordValue common.Hash, confs Confirmations) ([]Log, error)
 }
 
+// GetLogsBatchElem hides away all the interface casting, so the fields can be accessed more easily, and with type safety
+type GetLogsBatchElem rpc.BatchElem
+
+func NewGetLogsReq(filter Filter) *GetLogsBatchElem {
+	topics := make2DTopics(filter.EventSigs, filter.Topic2, filter.Topic3, filter.Topic4)
+
+	params := map[string]interface{}{
+		"address": []common.Address(filter.Addresses),
+		"topics":  topics,
+	}
+
+	return &GetLogsBatchElem{
+		Method: "eth_getLogs",
+		Args:   []interface{}{params},
+		Result: new([]types.Log),
+	}
+}
+
+func (e GetLogsBatchElem) params() map[string]interface{} {
+	return e.Args[0].(map[string]interface{})
+}
+
+func (e GetLogsBatchElem) Addresses() []common.Address {
+	return e.params()["address"].([]common.Address)
+}
+
+func (e GetLogsBatchElem) SetAddresses(addresses []common.Address) {
+	e.params()["address"] = addresses
+}
+
+func (e GetLogsBatchElem) Topics() [][]common.Hash {
+	return e.params()["topics"].([][]common.Hash)
+}
+
+func (e GetLogsBatchElem) SetTopics(topics [][]common.Hash) {
+	e.params()["topics"] = topics
+}
+
+func (e GetLogsBatchElem) FromBlock() *big.Int {
+	fromBlock, ok := e.params()["fromBlock"].(*big.Int)
+	if !ok {
+		return nil
+	}
+	return fromBlock
+}
+
+func (e GetLogsBatchElem) ToBlock() *big.Int {
+	toBlock, ok := e.params()["fromBlock"].(*big.Int)
+	if !ok {
+		return nil
+	}
+	return toBlock
+}
+
+func (e GetLogsBatchElem) BlockHash() *common.Hash {
+	blockHash, ok := e.params()["blockHash"].(*common.Hash)
+	if !ok {
+		return nil
+	}
+	return blockHash
+}
+
+func (e GetLogsBatchElem) SetFromBlock(fromBlock *big.Int) {
```

Comment:
`SetFromBlock` and `SetTopics` are probably not used, maybe not worth adding at this point.

Reply:
Yeah, I think they were in an earlier draft and I ended up not needing them.